delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/client/SparqlClient.scala

```diff
@@ -80,17 +80,18 @@ class SparqlClient(client: HttpClient, endpoint: SparqlQueryEndpoint)(implicit
     * @return
     *   successful Future[Unit] if update succeeded, failure otherwise
     */
-  def bulk(index: String, queries: Seq[SparqlWriteQuery]): IO[SparqlClientError, Unit] = {
+  def bulk(index: String, queries: Seq[SparqlWriteQuery]): IO[Unit] = {
     val queryString = queries.map(_.value).mkString("\n")
     val pss         = new ParameterizedSparqlString
     pss.setCommandText(queryString)
     for {
-      _          <- IO.fromTry(Try(pss.asUpdate())).mapError(th => InvalidUpdateRequest(index, queryString, th.getMessage))
+      _          <- IO(pss.asUpdate()).adaptError(e => InvalidUpdateRequest(index, queryString, e.getMessage))
```

Comment:
If `.apply` is the same as `.delay`, we probably want to decide which one to use. I know most existing code would use `delay`

Reply:
I'd go for apply, and not _just_ because fewer characters 😛  it follows the same pattern as `Try` and `Future`

Reply:
Personally I like the explicit nature and consistency with `.pure` that `.delay` has. But it's probably a team decision